### PR TITLE
communities: add request review workflow 

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

**Comment by ntarocco** [comment]
## To do

* tests all the scenario:
  - [ ] preselect the community via URL and submit
  - [ ] select community and save draft
  - [ ] select community and preview
  - [ ] select community and submit review
  - [ ] deselect community after save draft
  - [ ] edit a published record and check if community can/should changed
  - [ ] submit a review, then decline it, and see what happens to the record
* add a test for the `computeCommunityField` function

File: src/lib/DepositDraftsService.js [hunk 1]
```diff
@@ -1,16 +1,14 @@
 // This file is part of React-Invenio-Deposit
-// Copyright (C) 2020-2022 CERN.
-// Copyright (C) 2020-2022 Northwestern University.
-//
-// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
-// under the terms of the MIT License; see LICENSE file for more details.
```
Comment (Pineirin) [inline src/lib/DepositDraftsService.js:6]: Why removing doc?

File: src/lib/components/Communities/CommunityHeader.js [hunk 2]
```diff
@@ -1,68 +1,84 @@
...
+
+const mapDispatchToProps = (dispatch) => ({
+  changeSelectedCommunity: (community) =>
+    dispatch(changeSelectedCommunity(community)),
+});
+
+export const CommunityHeader = connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(CommunityHeaderComponent);
```
Comment (Pineirin) [inline src/lib/components/Communities/CommunityHeader.js:84]: Usually we do this part in a separate index.js file. Wouldn't it be more consistent to also do it here?

File: src/lib/components/FormFeedback.js [hunk 3]
```diff
@@ -79,6 +70,12 @@ const ACTIONS = {
       'Oops, something went wrong! The draft was not published. Please try again. If the problem persists, contact user support.'
     ),
   },
+  [DRAFT_SUBMIT_REVIEW_FAILED]: {
+    feedback: 'negative',
+    message: i18next.t(
+      'Oops, something went wrong! The draft was not submitted for review. Please try again. If the problem persists, contact user support.'
+    ),
+  },
```
Comment (Pineirin) [inline src/lib/components/FormFeedback.js:78]: Missing translation?

File: src/lib/components/Communities/CommunityHeader.js [hunk 4]
```diff
@@ -1,68 +1,84 @@
...
 
 CommunityHeaderComponent.propTypes = {
   imagePlaceholderLink: PropTypes.string.isRequired,
-  community: PropTypes.object.isRequired,
-  setCommunity: PropTypes.func.isRequired,
+  community: PropTypes.shape({
+    uuid: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    logo: PropTypes.string.isRequired,
+  }),
```
Comment (jrcastro2) [inline src/lib/components/Communities/CommunityHeader.js:69]: Silly question: Where is this data filled in?
Community doesn't have title outside the metadata and I see the following definiton below ([code here](https://github.com/inveniosoftware/react-invenio-deposit/pull/451/files#diff-48a39cc50930f6681605ecbd7bf7f66430fc8d3dde2d16292ec629476c25dce1R60-R73)):
```
  if (_selectedCommunity) {
    _selectedCommunity = {
      id: _selectedCommunity.id,
      uuid: _selectedCommunity.uuid || _selectedCommunity,
      metadata: {
        title: _selectedCommunity.title,
        description: _selectedCommunity.description,
        type: 'Type',
      },
      links: {
        self_html: '/',
      },
    };
  }
```

**Review by jrcastro2 — APPROVED** [review]
LGTM!

File: src/lib/components/Communities/CommunityHeader.js (hunk 2 above)
Comment (ntarocco) [reply src/lib/components/Communities/CommunityHeader.js:84]: IMO, it is annoying to have it in another file... I cannot see immediately what `props` are injected...

File: src/lib/components/FormFeedback.js (hunk 3 above)
Comment (ntarocco) [reply src/lib/components/FormFeedback.js:78]: Yes indeed, thanks I will add.

File: src/lib/components/PublishButton/SubmitReviewButton.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          )}
+        </ActionButton>
+        {isConfirmModalOpen && (
+          <Modal
+            open={isConfirmModalOpen}
+            onClose={this.closeConfirmModal}
+            size="small"
+          >
+            <Modal.Content>
+              <Trans>
```
Comment (kpsherva) [inline src/lib/components/PublishButton/SubmitReviewButton.js:76]: I think its missing the prop for <Trans> component, see 
https://react.i18next.com/latest/trans-component#using-with-react-components

File: src/lib/DepositApiClient.js [hunk 6]
```diff
@@ -188,6 +216,63 @@ export class RDMDepositApiClient extends DepositApiClient {
...
+   *
+   * @param {object} draftLinks - the draft links object
+   */
+  async submitReview(draftLinks) {
+    return this._createResponse(() =>
+      axiosWithConfig.post(
+        draftLinks['submit-review'],
+        {},
+        {
+          headers: { 'Content-Type': 'application/json' },
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply src/lib/DepositApiClient.js:271]: Maybe :) ping @zzacharo

**Pull request merged**